packages/mdc-ripple/README.md

```diff
@@ -61,15 +61,38 @@ General notes:
 
 #### Sass API
 
+In order to fully style states as well as the ripple effect for pressed state, both `mdc-ripple` mixins below must be included, as well as either the basic `mdc-states-color` mixin or all of the advanced `mdc-states` mixins documented below.
+
+Once these styles are in place for a component, it is feasible to further override only the parts necessary (e.g. `mdc-states-color` specifically) for specific variants (e.g. for flat vs. raised buttons).
+
 These APIs implicitly use pseudo-elements for the ripple effect: `::before` for the background, and `::after` for the foreground.
-All three of the following mixins are mandatory in order to fully style the ripple effect; from that point, it is feasible to further override only the parts necessary (e.g. `mdc-ripple-color` specifically) for variants of a component.
+
+##### Ripple Mixins
 
 Mixin | Description
 --- | ---
 `mdc-ripple-surface` | Adds base styles for a ripple surface
-`mdc-ripple-color($color, $opacity)` | Adds styles for the color and opacity of the ripple effect
 `mdc-ripple-radius($radius)` | Adds styles for the radius of the ripple effect,<br>for both bounded and unbounded ripples
 
+##### Basic States Mixin
+
+Mixin | Description
+--- | ---
+`mdc-states-color($color, $include-focus-within)` | Adds state and ripple styles for the indicated color, deciding opacities based on whether the passed color is light or dark. Optionally includes `::focus-within` selector for focused state of CSS-only components.
```

Comment:
We shouldn't document the internals of `$include-focus-within` as adding `::focus-within` selector. Thats an internal implementation detail. Could you turn this argument into '$css-only-component'?

Reply:
This isn't a blanket thing for all CSS-only components, and these mixins are only invoked once to cover both CSS-only and JS components. The distinction is that `::focus-within` is only needed for CSS-only components because we still handle focus/blur in JS when it is used in order to support all browsers (`::focus-within` isn't supported in IE/Edge).

Your comment later about `$focusable-element-within-root-element` could apply equally here; this is simply forwarding to that parameter when invoking the focus mixin. Maybe `$has-nested-focusable-element`? Coming up with a short name for this is hard...

Reply:
$has-nested-focusable-element is better. Its okay it is long.